### PR TITLE
Use default user shipping address to calculate indicative shipping cost on basket summary page.

### LINKED ISSUE #827: Extend basket view to use default shipping address
The basket summary view attempts to look up a default shipping method so an indicative order total can be displayed.  The checkout allows the method to be changed later on though.

At the basket stage, we don't know the user's chosen shipping address so we don't pass it.  However, Oscar does have a concept of default shipping address.  We should extend the basket view to pass the user's default shipping address (if one is known).

### PR DESCRIPTION
**Pull request by sasha0**
Fixes #827

### CODE REVIEW

File: src/oscar/apps/basket/views.py [hunk 1]
```diff
@@ -54,10 +54,16 @@ def get_shipping_methods(self, basket):
             basket=self.request.basket, user=self.request.user,
             request=self.request)
 
+    def get_default_shipping_address(self):
+        if self.request.user.is_authenticated():
+            shipping_addresses = self.request.user.addresses.filter(is_default_for_shipping=True)
```
Comment (solarissmoke) [inline src/oscar/apps/basket/views.py]: Can do this all in one query: `user.addresses.filter(is_default_for_shipping=True).first()`, which will be `None` if none exists.

**Comment by solarissmoke** [comment]
The `get_default_shipping_address` method needs a unit test?

**Comment by sasha0** [comment]
Quite simple to live without it.

**Comment by solarissmoke** [comment]
I disagree - I think it's important even it if looks simple. Several bugs have come up recently after adding support for Django 1.11 and 2.0 because of untested code where upstream behaviour has changed. If we had proper test coverage then these would have been picked up earlier - a good example is the `AdvancedWidget` which had no tests for what seemed like a simple behaviour.

**Comment by sasha0** [comment]
Alright then, agreed.

File: src/oscar/apps/basket/views.py [hunk 2]
```diff
@@ -54,10 +54,15 @@ def get_shipping_methods(self, basket):
             basket=self.request.basket, user=self.request.user,
             request=self.request)
 
+    def get_default_shipping_address(self):
+        if self.request.user.is_authenticated():
```
Comment (solarissmoke) [inline src/oscar/apps/basket/views.py]: Should be using `core.compat.is_user_authenticated()` here - calling the method is deprecated.

**Comment by solarissmoke** [comment]
Will you add that unit test?

**Comment by sasha0** [comment]
Sure. Requested review again by mistake.

**Comment by sasha0** [comment]
@solarissmoke ok I've added tests.

**Comment by solarissmoke** [comment]
@sasha0 tests seem to be failing?

**Comment by sasha0** [comment]
@solarissmoke all fixed and ready for the review.

**Pull request merged**